### PR TITLE
MA-83 Universal Webhook Support

### PR DESCRIPTION
**Pull request by njfaries**
# Description
This PR goes along with https://github.com/BoltApp/bolt-magento2/pull/1085. This adds a webhook endpoint for V2 webhooks to be sent to and simply pulls the data needed to pass to the existing `OrderManagement::manage` method and returns either a success response or the error thrown by `OrderManagement::manage`

The inclusion of the result interface is a bit excessive for now as we are continuing to create and send the responses manually as opposed to returning the interface to the Magento REST handler. I'm including it here as in https://boltpay.atlassian.net/browse/MA-493 I hope to clean up the response handling to make better use of interfaces if we still cannot simply return interfaces to the Magento REST handler.

[Relevant slack discussion](https://boltpay.slack.com/archives/CQ864R9JQ/p1610377344014400)

Fixes: https://boltpay.atlassian.net/browse/MA-83

#changelog MA-83 Universal Webhook Support
Support for V2 Universal Webhook

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: Model/Api/UniversalWebhook.php [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+     */
+    public function execute(
+        $type = null,
+        $object = null,
+        $data = null
+    )
+    {
+        try {
+            // This just reuses the existing manage function by pulling the necessary data from the request.
+            $this->orderManagement->manage(
```
Comment (vitaliyreznikov) [inline Model/Api/UniversalWebhook.php:96]: The purpose of the orderManagement class is to handle the specific API endpoint.
If we need the same code for handling two different API endpoints I'd rather prefer to move the code to the third class and call it from two API classes then call one API from another.

Comment (njfaries) [reply Model/Api/UniversalWebhook.php:96]: While I don't disagree with your point here, this is already what's been done in https://github.com/BoltApp/bolt-magento2/pull/1085 now: calling existing handler methods from the universal handler. Do you think this is something that could be scoped as a refactor after getting the MVP version of the universal webhook and api out? Or should this block merging altogether and we set up tickets to refactor the universal API stuff as well?

Comment (vitaliyreznikov) [reply Model/Api/UniversalWebhook.php:96]: I am ok to merge this, please add task to the backlog for refactoring

**Review by vitaliyreznikov — APPROVED** [review]
LGTM

**Pull request merged**